services/QuillLMS/client/app/bundles/Connect/components/focusPoints/focusPointsContainer.jsx

```diff
@@ -136,8 +143,20 @@ export class FocusPointsContainer extends Component {
     return <SortableList data={_.values(components)} key={_.values(components).length} sortCallback={this.sortCallback} />;
   }
 
-  renderTagsForFocusPoint = (focusPointString) => {
-    return focusPointString.split('|||').map((fp, index) => (<span className="tag is-medium is-light" key={`fp${index}`} style={{ margin: '3px', }}>{fp}</span>));
+  handleChange = (e, key) => {
+    const { focusPoints } = this.state
+    let value = e.target.value;
+    let className = `regex-${key}`
+    value = `${Array.from(document.getElementsByClassName(className)).map(i => i.value).filter(val => val !== '').join('|||')}`;
+    focusPoints[key].text = value;
+    this.setState({focusPoints: focusPoints})
+  }
```

Comment:
This is small, but ideally state-change functions all go above `render` functions.